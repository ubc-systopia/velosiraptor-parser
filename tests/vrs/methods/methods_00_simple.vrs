segment MyUnit {
    // method without body
    fn foo() -> bool

    // method without body with semicolon
    fn foo() -> bool;

    // method with a body
    fn foo() -> bool {
        true
    }

    // another method with a body
    fn bar() -> int {
        1 + 2
    }

    // method with an argument
    fn bar(val: int) -> int {
        1 + val
    }

    // method with multiple arguments
    fn bar(val: int, val2: bool) -> int {
        if val2 {
            1 + val
        } else {
            val
        }
    }
}