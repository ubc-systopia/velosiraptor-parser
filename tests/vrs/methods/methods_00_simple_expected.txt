VelosiParseTree(tests/vrs/methods/methods_00_simple.vrs)
---------------------------------------------
segment MyUnit {
  fn foo() -> bool

  fn foo() -> bool

  fn foo() -> bool
  {
    true
  }

  fn bar() -> int
  {
    1 + 2
  }

  fn bar(val: int) -> int
  {
    1 + val
  }

  fn bar(val: int, val2: bool) -> int
  {
    if val2 { 1 + val } else { val }
  }

}


---------------------------------------------
